location.md

```diff
@@ -0,0 +1,234 @@
+# Location 
+
+## Overview
+
+* A Location is a physical place where a patient may be seen, such as a hospital, a room, a clinic, or a district.
+
+* <b>Locations may have a hierarchy</b>,such that each location may have one parent location.
+
+* And also a Location can have one or more Children location example Children's Ward might be a location within the location 
+Amani Clinic.
+  
+* You might also store physical areas (for example Eastern Province, or California) as Locations. 
+
+* You should not use locations to represent logical ideas like All District Hospitals they should be modeled using LocationTags.
+
+## Sub Resource types
+
+### Attribute.
+
+* If you wish to record extra information about location, you can create Location Attributes and assign them to Location Types.
+
+* Location attributes exists specifically to allow implementations to extend the data model.
+
+
+## Available operations. 
+
+1. [List location](#list-location)
+2. [Create a location](#create-a-location)
+3. [Update a location](#update-a-location)
+4. [Delete a location](#delete-a-location)
+5. [List location attribute sub resource](#list-location-attribute-sub-resources)
+6. [Create location attribute sub resource with properties](#create-a-location-attribute-sub-resource-with-properties)
+7. [Update location attribute sub resource](#update-a-location-attribute-sub-resource)
+6. [Delete location attribute sub resource](#delete-a-location-attribute-sub-resource)
+
+
+
+### List location
+
+* #### List all non-retired locations`.
+    
+    Quickly filter location with given query parameters.Returns a `404 Not Found` status if location not exists. If user not
+    logged in to perform this action,a `401 Unauthorized` status returned.
+    
+    ##### Query Parameters
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *q* | `Search Query` | Display Name of Location object.
+     
+    ```console
+     GET /location?
+     q="Search Query"
```

Comment:
URLs may not have spaces. Consider using an example, like `GET /location?q=amani`